Decorative block

MineShadow7 opened the pull request:
# Changes proposed in this pull request:
- 1 new Decorative block added
- Crafts for tools (Build tool -> Scan tool -> Permission tool)

Review please

MineShadow7 committed:
Added blocks and recipes for tools

MineShadow7 committed:
Merge remote-tracking branch 'origin/version/1.10' into DecorativeBlocksAndToolsCrafts

MineShadow7 committed:
constants changes

MineShadow7 committed:
Description changes

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockTimberFrame.java:
```diff
@@ -0,0 +1,68 @@
+     *
+     * @return true
+     */
+    //todo: remove once we no longer need to support this
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:
```diff
@@ -74,12 +75,15 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
                 "XXX", "X#X", "XXX", 'X', PLANK_WOOD, '#', ModItems.scanTool));
         GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockSolidSubstitution, ONE_FORTH_OF_A_STACK),
                 "XXX", "X#X", "XXX", 'X', "logWood", '#', ModItems.scanTool));
+        GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockTimberFrame, 4), "X", "X", "X", 'X', PLANK_WOOD));
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:
```diff
@@ -74,12 +75,15 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
         GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockSolidSubstitution, ONE_FORTH_OF_A_STACK),
                 "XXX", "X#X", "XXX", 'X', "logWood", '#', ModItems.scanTool));
+        GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockTimberFrame, 4), "X", "X", "X", 'X', PLANK_WOOD));
+        GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockCrossedFrame, 4), "X X", " X ", "X X", 'X', PLANK_WOOD));
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:
```diff
@@ -8,6 +8,7 @@
 import net.minecraft.init.Items;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.Mod;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused import 'net.minecraftforge.fml.common.Mod'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented:
SonarQube analysis reported 53 issues
* ![MAJOR][MAJOR] 19 major
* ![MINOR][MINOR] 30 minor
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.265 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L663](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java#L663): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L853](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java#L853): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1202](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1202): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1203](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1203): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1316](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1316): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1318](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1318): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1320](https://github.com/Minecolonies/minecolonies/blob/96a03d8c87463f982bd059910fc7413d6f4d9b49/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1320): Assign this magic number 128 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockCrossedFrame.java:
```diff
@@ -0,0 +1,68 @@
+     *
+     * @return true
+     */
+    //todo: remove once we no longer need to support this
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MineShadow7 committed:
Sonar

MineShadow7 committed:
Sonar Changes

MineShadow7 committed:
Sonar changes. Permission tool craft fixed.

MineShadow7 committed:
Added build Tool to frames craft to make them unique

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:
```diff
@@ -32,9 +32,12 @@
      */
     private static final int ONE_FORTH_OF_A_STACK = 16;
+
+    private static final int FOUR_BLOCKS = 4;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "FOUR_BLOCKS" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:
```diff
@@ -74,12 +77,16 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
                 "XXX", "X#X", "XXX", 'X', PLANK_WOOD, '#', ModItems.scanTool));
         GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockSolidSubstitution, ONE_FORTH_OF_A_STACK),
                 "XXX", "X#X", "XXX", 'X', "logWood", '#', ModItems.scanTool));
+        GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockTimberFrame, 4), "W", "W", "WB", 'W', PLANK_WOOD, 'B', ModItems.buildTool));
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:
```diff
@@ -74,12 +77,16 @@ public static void init(final boolean enableInDevelopmentFeatures, final boolean
         GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockSolidSubstitution, ONE_FORTH_OF_A_STACK),
                 "XXX", "X#X", "XXX", 'X', "logWood", '#', ModItems.scanTool));
+        GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockTimberFrame, 4), "W", "W", "WB", 'W', PLANK_WOOD, 'B', ModItems.buildTool));
+        GameRegistry.addRecipe(new ShapedOreRecipe(new ItemStack(ModBlocks.blockCrossedFrame, 4), "W W", " W ", "WBW", 'W', PLANK_WOOD, 'B', ModItems.buildTool));
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented:
SonarQube analysis reported 52 issues
* ![MAJOR][MAJOR] 20 major
* ![MINOR][MINOR] 29 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.265 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L663](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L663): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L853](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L853): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1202](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1202): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1203](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1203): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1316](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1316): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1318](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1318): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1320](https://github.com/Minecolonies/minecolonies/blob/cf10f4bfbd360aa14ec9ad3ad7b925f1327bea4b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1320): Assign this magic number 128 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

MineShadow7 committed:
Using constants for now. No magic numbers...

MineShadow7 committed:
Merge remote-tracking branch 'origin/DecorativeBlocksAndToolsCrafts' into DecorativeBlocksAndToolsCrafts

MineShadow7 committed:
And another fix to crashing and added possibility to craft scan tool from permission tool.

MinecoloniesSonar commented:
SonarQube analysis reported 49 issues
* ![MAJOR][MAJOR] 17 major
* ![MINOR][MINOR] 29 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.265 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L663](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L663): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L853](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L853): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1202](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1202): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1203](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1203): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1316](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1316): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1318](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1318): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1320](https://github.com/Minecolonies/minecolonies/blob/07a4a225c6b36a2109bffb6799532421d336718b/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1320): Assign this magic number 128 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

MineShadow7 commented:
If no errors appear - it's finally done.

xavierh commented:
Would you mind to attached screen shots of the new blocks ?

MineShadow7 commented:
Sure
![2017-06-20_14 07 22](https://user-images.githubusercontent.com/28764927/27444550-656168b6-577f-11e7-88cf-c2ac25a4b274.png)
![2017-06-22_19 17 18](https://user-images.githubusercontent.com/28764927/27444551-657cc3ae-577f-11e7-8f86-8146b067fd46.png)

marchermans commented on src/main/java/com/minecolonies/coremod/blocks/BlockTimberFrame.java:
```diff
@@ -0,0 +1,67 @@
+/**
+ * Decorative blocks
+ */
+public class BlockTimberFrame extends Block
```
Why is this a different Block and not just a different BlockState.
This is occupying an additional BlockID while it is not needed nor recommended,

marchermans commented on src/main/java/com/minecolonies/coremod/colony/Colony.java:
```diff
@@ -820,7 +820,10 @@ public void onWorldUnload(@NotNull final World w)
+            /**
+             * It's not very good to crash here. We should simply log this so we can remove this colony.
+             */
+            Log.getLogger().error("Colony %d has the wrong world, colony probably should be removed if this is being spammed.", id);
```
Why was this changed in this PR?
Please do not Change this.

MineShadow7 replied:
I was not changing it here myself...

marchermans reviewed (changes requested):
Use 1 Block for all of the DecoBlocks.
And do not Combine fixes with Features.

MineShadow7 commented:
Okay, this branch needs some update:
Better solution to this is remove one unnecessary block to make this easier and possible.

MineShadow7 committed:
Removed Crossed Frame

MinecoloniesSonar commented:
SonarQube analysis reported 48 issues
* ![MAJOR][MAJOR] 17 major
* ![MINOR][MINOR] 29 minor
* ![INFO][INFO] 2 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.265 lines, which is greater than 1.000 authorized. Split it into smaller files. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L47](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java#L47): class "Colony" has 84 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L663](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java#L663): Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L853](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java#L853): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1202](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1202): Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1203](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1203): Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1316](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1316): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1318](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1318): Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1320](https://github.com/Minecolonies/minecolonies/blob/14ceccf711f58fa0030085327e592959ae3b44ec/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1320): Assign this magic number 128 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

MineShadow7 committed:
Merge latest 1.10 to this branch to keep it fresh

MinecoloniesSonar commented:
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 28 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L102](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L102): Remove this unused private "enableInDevelopmentFeatures" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L121](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L121): Remove this unused private "addHutRecipe" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [ModBlocks.java#L38](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java#L38): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/578ae8b2f4b7fec81a1144fa5117e8b07cce1ea1/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

MineShadow7 commented:
Now finally ready for review.

Asherslab commented:
Looks good to me :shipit: @OrionDevelopment @Raycoms 

[![Approved with PullApprove](https://img.shields.io/badge/maintainers-approved-brightgreen.svg)](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1222/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

MineShadow7 committed:
*New texture by d3miurge.

*Changed block hardness to make more "wooden"

MinecoloniesSonar commented:
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 28 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L102](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L102): Remove this unused private "enableInDevelopmentFeatures" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L121](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L121): Remove this unused private "addHutRecipe" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [ModBlocks.java#L38](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java#L38): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/1d8f59c05c4df6304e773aeaa630a42c4a961c4d/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

MineShadow7 commented:
![2017-08-02_23 59 57](https://user-images.githubusercontent.com/28764927/28894969-3f5c3c60-77df-11e7-8eaf-6a5f15974726.png)
![2017-08-03_13 51 31](https://user-images.githubusercontent.com/28764927/28918740-476460e4-7853-11e7-91a7-ee3813b1a31f.png)




New textures by d3miurge

MineShadow7 committed:
Improved up and down sides of block

MinecoloniesSonar commented:
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 28 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L102](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L102): Remove this unused private "enableInDevelopmentFeatures" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L121](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L121): Remove this unused private "addHutRecipe" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [ModBlocks.java#L38](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java#L38): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/ad1d032fc426072fcf5fd72e1b0c91bb491b18f2/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

MineShadow7 commented:
Anyone?

Asherslab commented:
@OrionDevelopment Need a new review on this please =D :shipit:

Raycoms commented:
:shipit: 

[![Approved with PullApprove](https://img.shields.io/badge/maintainers-approved-brightgreen.svg)](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1222/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

Raycoms committed:
Merge branch 'version/1.10' into DecorativeBlocksAndToolsCrafts

MinecoloniesSonar commented:
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 28 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L102](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L102): Remove this unused private "enableInDevelopmentFeatures" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L121](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L121): Remove this unused private "addHutRecipe" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [ModBlocks.java#L38](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java#L38): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/a51a78b06171a81f67571421f284a3623297cd9e/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Asherslab committed:
Merge branch 'version/1.10' into DecorativeBlocksAndToolsCrafts

MinecoloniesSonar commented:
SonarQube analysis reported 35 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 33 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L102](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L102): Remove this unused private "enableInDevelopmentFeatures" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L121](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L121): Remove this unused private "addHutRecipe" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [ModBlocks.java#L38](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java#L38): This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [requestsystem](https://github.com/Minecolonies/minecolonies/blob/4c736e654b0fb17355b2c5f59e90c57f9a4be2a4/src/main/java/com/minecolonies/coremod/colony/requestsystem): Add a 'package-info.java' file to document the 'requestsystem' package [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Asherslab commented:
Can you port this to version/1.11 and version/1.12.

Also, anychance you can incorporate these decorative blocks into some of the Minecolonies schematics?

MineShadow7 commented:
I can do that. But should I do it myself if our best builders can do it better?

Asherslab commented:
I was meaning more, could you organise that, either get another person (Say D3 prob) and make a PR with them, or whatever you choose. I'm simply saying could you do or Delegate that
